tools/toolchain_profiler.py

```diff
@@ -4,19 +4,35 @@
 # found in the LICENSE file.
 
 import atexit
-import subprocess
+import logging
 import os
-import time
 import sys
+import subprocess
 import tempfile
+import time
 from contextlib import ContextDecorator
 
 sys.path.insert(1, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+logger = logging.getLogger('profiler')
 
 from tools import response_file
 
 EMPROFILE = int(os.getenv('EMPROFILE', '0'))
 
+
+class Logger(ContextDecorator):
```

Comment:
Is this class used by anything except `ProfileBlock`? If not, is it worth keeping separate?

Reply:
Its used twice below.. once in the EMPROFILE case (where profiling is enabled) and also in the null case.. when we still want the simple logging but no the full profiling setup.

Reply:
Oh, I see, thanks!